File: src/classes/api/req_res/upload-gps-log-files-request.ts
'use strict'

import { FileData } from '../file-data'
import { GkillAPIRequest } from '../gkill-api-request'
import { FileUploadConflictBehavior } from './file-upload-conflict-behavior'

export class UploadGPSLogFilesRequest extends GkillAPIRequest {

    gps_log_files: Array<FileData>

    target_rep_name: string

    conflict_behavior: FileUploadConflictBehavior

    constructor() {
        super()
        this.gps_log_files = new Array<FileData>()
        this.target_rep_name = ""
        this.conflict_behavior = FileUploadConflictBehavior.merge
    }

}

File: src/app/gkill/api/req_res/upload_gps_log_files_request.go
package req_res

type UploadGPSLogFilesRequest struct {
	SessionID string `json:"session_id"`

	GPSLogFiles []*FileData `json:"gps_log_files"`

	TargetRepName string `json:"target_rep_name"`

	ConflictBehavior FileUploadConflictBehavior `json:"conflict_behavior"`
}
